frontend/src/pages/upload/UploadPage.tsx

```diff
@@ -118,34 +144,57 @@ const UploadPage: FunctionComponent = (): ReactElement => {
             )}
           </FormField>
 
-          <FormField
-            as="textarea"
-            name="description"
-            label="Description"
-            className="uploadingForm__description"
-            classNameInner="ednaTextarea"
-            tabIndex={4}
-          />
+          {isParsed(excelFile) && (
+            <div className="uploadingForm__label">* marked will be created</div>
+          )}
+
+          <div className="uploadingForm__buttons">
+            <Button
+              disabled={!isParsed(excelFile)}
+              type="submit"
+              className="uploadingForm__submitBtn"
+              tabIndex={4}
+            >
+              Add experiments
+            </Button>
 
-          <button className="primaryButton uploadingForm__submitBtn" type="submit" tabIndex={5}>
-            Save
-          </button>
+            <Button
+              type="outlined"
+              className="uploadingForm__anotherBtn"
+              tabIndex={5}
+              onClick={() => {
+                setExcelFile(undefined);
+              }}
+            >
+              Upload another one
+            </Button>
+          </div>
         </Form>
       </Formik>
 
       <div className="uploadResult">
-        {isSucceeded(uploadingStatus) && (
-          <MeasurementsCharts measurements={uploadingStatus.result} />
-        )}
+        {isParsed(excelFile) && <MeasurementsCharts experiments={excelFile.experiments} />}
 
-        {uploadingStatus.status === "loading" && <div className="spinner">Uploading...</div>}
+        {(excelFile?.state === "uploading" || excelFile?.state === "verifying") && (
+          <div className="spinner">Uploading...</div>
+        )}
 
-        {uploadingStatus.status === "failed" && (
-          <span className="uploadResult__fail">{uploadingStatus.error}</span>
+        {(excelFile?.state === "failed-to-parse" || excelFile?.state === "failed-to-add") && (
+          <span className="uploadResult__fail">{excelFile.reason}</span>
         )}
       </div>
     </PageLayout>
   );
 };
 
-export default UploadPage;
+function uploadFormToApi(form: UploadForm): Maybe<UploadExperimentsArgs> {
```

Comment:
Maybe move it to utils on put it in the beginning of the component to see it before the place where it is executing

Reply:
Maybe move your to detdom for this kind of suggestions? :putya: